web-common-marlowe/src/Marlowe/Extended/Metadata.purs

```diff
@@ -35,13 +35,51 @@ instance showChoiceFormat :: Show ChoiceFormat where
   show = genericShow
 
 integerFormat :: ChoiceFormat
-integerFormat = Decimal 0 ""
+integerFormat = DecimalFormat 0 ""
 
 lovelaceFormat :: ChoiceFormat
-lovelaceFormat = Decimal 6 "₳"
+lovelaceFormat = DecimalFormat 6 "₳"
 
 oracleRatioFormat :: String -> ChoiceFormat
-oracleRatioFormat str = Decimal 8 str
+oracleRatioFormat str = DecimalFormat 8 str
+
+isDefaultFormat :: ChoiceFormat -> Boolean
+isDefaultFormat DefaultFormat = true
+
+isDefaultFormat _ = false
+
+isDecimalFormat :: ChoiceFormat -> Boolean
+isDecimalFormat (DecimalFormat _ _) = true
+
+isDecimalFormat _ = false
```

Comment:
with the changes suggested above, we could get rid of this boilerplate 🤷

Reply:
It would be a fair approach (I did consider it too), but I almost prefer it with the boilerplate in this case. I justify it to myself in that we may reuse them, and in that it is more concise (in View.purs) and slightly more efficient (even though, of course, it would be completely imperceptible)